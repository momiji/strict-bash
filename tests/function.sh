function f(){ }
function f{ }
f() { }
