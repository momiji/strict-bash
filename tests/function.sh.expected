start: State(name='start', start=0, end=0, depth=0)
    start: State(name='statement', start=0, end=0, depth=1)
        start: State(name='command', start=0, end=0, depth=2)
            start: State(name='arg', start=0, end=0, depth=3)
            end: State(name='arg', start=0, end=8, depth=3) => 'function'
            start: State(name='arg', start=9, end=9, depth=3)
            end: State(name='arg', start=9, end=12, depth=3) => 'f()'
            start: State(name='{', start=12, end=12, depth=3)
                start: State(name='statement', start=13, end=13, depth=4)
                end: State(name='statement', start=13, end=14, depth=4) => ' '
            end: State(name='{', start=12, end=15, depth=3) => '{ }'
        end: State(name='command', start=0, end=15, depth=2) => 'function f(){ }'
    end: State(name='statement', start=0, end=15, depth=1) => 'function f(){ }'
    start: State(name='statement', start=16, end=16, depth=1)
        start: State(name='command', start=16, end=16, depth=2)
            start: State(name='arg', start=16, end=16, depth=3)
            end: State(name='arg', start=16, end=24, depth=3) => 'function'
            start: State(name='arg', start=25, end=25, depth=3)
            end: State(name='arg', start=25, end=26, depth=3) => 'f'
            start: State(name='{', start=26, end=26, depth=3)
                start: State(name='statement', start=27, end=27, depth=4)
                end: State(name='statement', start=27, end=28, depth=4) => ' '
            end: State(name='{', start=26, end=29, depth=3) => '{ }'
        end: State(name='command', start=16, end=29, depth=2) => 'function f{ }'
    end: State(name='statement', start=16, end=29, depth=1) => 'function f{ }'
    start: State(name='statement', start=30, end=30, depth=1)
        start: State(name='command', start=30, end=30, depth=2)
            start: State(name='arg', start=30, end=30, depth=3)
            end: State(name='arg', start=30, end=33, depth=3) => 'f()'
            start: State(name='{', start=34, end=34, depth=3)
                start: State(name='statement', start=35, end=35, depth=4)
                end: State(name='statement', start=35, end=36, depth=4) => ' '
            end: State(name='{', start=34, end=37, depth=3) => '{ }'
        end: State(name='command', start=30, end=37, depth=2) => 'f() { }'
    end: State(name='statement', start=30, end=37, depth=1) => 'f() { }'
end: State(name='start', start=0, end=38, depth=0) => 'function f(){ }\nfunction f{ }\nf() { }\n'
